src/statd/statd.c

```diff
@@ -256,6 +258,37 @@ static int sr_routes_cb(sr_session_ctx_t *session, uint32_t, const char *path,
 	return err;
 }
 
+static int sr_ospf_cb(sr_session_ctx_t *session, uint32_t, const char *path,
+			const char *, const char *, uint32_t,
+			struct lyd_node **parent, __attribute__((unused)) void *priv)
+{
+	const struct ly_ctx *ctx;
+	sr_conn_ctx_t *con;
+	sr_error_t err;
+
+	DEBUG("Incoming query for xpath: %s", path);
+
+	con = sr_session_get_connection(session);
+	if (!con) {
+		ERROR("Error, getting sr connection");
+		return SR_ERR_INTERNAL;
+	}
+
+	ctx = sr_acquire_context(con);
+	if (!ctx) {
+		ERROR("Error, acquiring context");
+		return SR_ERR_INTERNAL;
+	}
```

Comment:
You're leaking the `con` here, and I'm noticing now there are more of the same in previous functions. 

I suggest using the same pattern as in the kernel, `if (func()) { err = CODE; goto end;}`, where `end:` releases all memory and returns err?

Reply:
The documentation for sr_session_get_connection say:
<pre>
Get the connection the session was created on.
</pre>

I interpret it as I should not free it in this context.

Reply:
Was just an observation since you call `sr_release_context(con)` a few lines down, but not for the first error case (in all these callbacks).